### ISSUE TITLE
Transitive reference to System.Configuration.ConfigurationManager doesn't work in net6.0 project

### ISSUE THREAD

**Issue by benvillalobos** [NuGet/Home#11692]
---
**Issue moved from dotnet/msbuild#7486**
- Please respond to @madelson.

---

_From @madelson on Wednesday, March 23, 2022 9:42:46 PM_

**Version Used**:  VS 17.0.1

**Steps to Reproduce**:

1. In VS, create a new net6.0 class library project. Set the csproj to the following:

```XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="System.Runtime.Caching" Version="4.7.0" />
    <!--<PackageReference Include="System.Configuration.ConfigurationManager" Version="4.7.0" />-->
  </ItemGroup>
</Project>
```

Upon compiling, observe the `System.Runtime.Configuration.ConfigurationManager` is pulled in transitively in packages.lock.json:

```
      "System.Configuration.ConfigurationManager": {
        "type": "Transitive",
        "resolved": "4.7.0",
        "contentHash": "/anOTeSZCNNI2zDilogWrZ8pNqCmYbzGNexUnNhjW8k0sHqEZ2nHJBp147jBV3hGYswu5lINpNg1vxR7bnqvVA==",
        "dependencies": {
          "System.Security.Cryptography.ProtectedData": "4.7.0",
          "System.Security.Permissions": "4.7.0"
        }
      },
```

2.  Add a reference to `ConfigurationManager` in the C# source (Class1.cs):

```C#
namespace ClassLibrary1
{
	public class Class1
	{
		public Type Method()
		{
			return typeof(System.Configuration.ConfigurationManager);
		}
	}
}
```

3. Compile.
 
**Expected Behavior**:

Compiles without error

**Actual Behavior**:

```
error CS1069: The type name 'ConfigurationManager' could not be found in the namespace 'System.Configuration'. This type has been forwarded to assembly 'System.Configuration.ConfigurationManager, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' Consider adding a reference to that assembly.
```

Note that this can be fixed by uncommenting the direct reference to the configuration manager package. But why should this be necessary?

**Comment by benvillalobos** [comment]
---
**Issue moved from dotnet/msbuild#7486**
- Please respond to @madelson.

---

_From @madelson on Thursday, March 24, 2022 12:54:41 PM_

@jcouv can you explain why this is labeled as "question"? Is this behavior expected?

**Comment by benvillalobos** [comment]
---
**Issue moved from dotnet/msbuild#7486**
- Please respond to @jcouv.

---

_From @jcouv on Thursday, March 24, 2022 4:12:55 PM_

I'll let msbuild folks triage and confirm whether question or bug.

**Comment by benvillalobos** [comment]
---
**Issue moved from dotnet/msbuild#7486**
- Please respond to @BenVillalobos.

---

_From @BenVillalobos on Thursday, March 24, 2022 4:40:47 PM_

MSBuild Team Triage: This looks like a NuGet issue. We repro'd by:

`dotnet new console`
Bring in 
```
  <ItemGroup>
    <PackageReference Include="System.Runtime.Caching" Version="4.7.0" />
    <!--<PackageReference Include="System.Configuration.ConfigurationManager" Version="4.7.0" />-->
  </ItemGroup>
```
Restore
Build
It repro's.
Uncomment the other packagereference and it will work.

We diff'd the before/after `projects.assets.json`, we saw weird items like 
![image](https://user-images.githubusercontent.com/4691428/159966407-4723b28a-a272-4d2a-943e-05a6580f4caa.png)

Forwarding to nuget

**Comment by nkolev92** [comment]
The `System.Runtime.Caching` package is authored with: 

>         <dependency id="System.Configuration.ConfigurationManager" version="4.7.0" exclude="Compile" />

Which is why you don't get the compile assets from `System.Configuration.ConfigurationManager`. 

That's normal and it's how the package author intended it. 
If you want to depend on `System.Configuration.ConfigurationManager` at compile time, they want you to take a direct dependency. 

tldr; 
This a package authoring decision, and there's no tooling issues.

**Comment by madelson** [comment]
Interesting @nkolev92 . Thanks for the explanation!

**Comment by madelson** [comment]
@nkolev92 is there documentation somewhere on how the include/exclude tags on a package's dependency element interact with restore? If a package appears multiple times in the restore graph, which sets of tags gets applied on the final compilation? Is in the union of all the (include - exclude) sets across all references to that package in the tree?
